### ISSUE TITLE
Spurious http 400 errors (possible rate limits)

### ISSUE THREAD

**Issue by ivandeex** [ivandeex/rclone#14]
- [x] add debug option to print ALL errors from server
- [x] add debug option to retry on 400
- [x] describe workaround in docs

**Comment by ivandeex** [comment]
**Problem description**
During development I once ran "rclone ls" against a deep directory tree and observed that subdirectory lists were randomly truncated, and the command at last failed with a http error. Location of errors seemed indeterminate at first, so I looked deeper into messages returned by server, got to status 400 and hardcoded it as a retry code to investigate its behavior. The errors were spurious with probability dependant on directory depth, so I tried throttling with `--tpslimit 2`. Then I noticed that another rclone mount had been running for ages on another machine. Killing it and restarting the test with re-authorization and TPS limiting dismissed the errors. After a while the problem has gone competely. Probably mailru maintains undocumented throttling limits and parallel access made it hit.

**Bottom line**
Similar problems are possible in future. They are hard to troubleshoot, especially for remote users. The backend needs a maintenance flag (e.g. `dumperr`) to help with isolating a root cause. The `--tpslimit` cli flag might be considered as a first-aid solution. Also the `retry400` quirk will be added.
